Scripts/RemediationScripts/Remediate-StopExposingServiceFabricReverseProxyPort.ps1

```diff
@@ -0,0 +1,858 @@
+
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+
+        Write-Host "Fetching all Service Fabric(s) from [$($FilePath)]..." 
+
+        $ServiceFabricResources = Import-Csv -LiteralPath $FilePath
+        $validServiceFabricResources = $ServiceFabricResources| Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+        
+        $validServiceFabricResources| ForEach-Object {
+            $resourceId = $_.ResourceId
+
+            try
+            {                
+                $ServiceFabricResource =  Get-AzServiceFabricCluster -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+            
+                $ServiceFabricDetails += $ServiceFabricResource  | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                                          @{N='ResourceGroupName';E={$_.Id.Split("/")[4]}},
+                                                                          @{N='ResourceName';E={$_.Name}},
+                                                                          @{N='IsReverseProxyPortEnabled';E={$_.NodeTypes.ReverseProxyEndpointPort.Count -gt 0}},
+                                                                          @{N='ReverseProxyPorts';E={if($_.NodeTypes.ReverseProxyEndpointPort.Count -gt 0)
+                                                                          { 
+                                                                                $_.NodeTypes.ReverseProxyEndpointPort
+                                                                          }                                                                          
+                                                                          }}
+            }
+            catch
+            {
+                Write-Host "Error fetching Service Fabric(s) resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }
+        }
+    }
+
+    $totalServiceFabric = ($ServiceFabricDetails| Measure-Object).Count
+
+    if ($totalServiceFabric -eq 0)
+    {
+        Write-Host "No Service Fabric(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        break
+    }
+  
+    Write-Host "Found [$($totalServiceFabric)] Service Fabric(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing Service Fabric(s) for which Reverse proxy port are exposed.
+    $ServiceFabricWithReverseProxyPortExposed = @()
+
+    Write-Host "Separating Service Fabric(s) for which Reverse proxy port are exposed..."
+     
+    $ServiceFabricDetails | ForEach-Object {
```

Comment:
Control is not applicable for ERvNET connected Service Fabric nodes, we should add check to remove such instances if that's complex then go with confirmation from user.

Reply:
Changes added to stop remediating the Service Fabric of ErVnet type